.github/workflows/lint_test.yml

```diff
@@ -0,0 +1,144 @@
+    inputs:
+      package:
+        description: 'Package to release'
+        required: true
+        type: choice
+        options:
+          - schematic-js
+          - schematic-react
+          - schematic-components
+          - schematic-vue
+      version:
+        description: 'RC version (e.g., 1.2.0-rc.0)'
+        required: true
+        type: string
+      dry_run:
+        description: 'Dry run (skip publish)'
+        required: false
+        type: boolean
+        default: false
+
+jobs:
+  validate:
+    name: Validate Input
+    runs-on: ubuntu-latest
+    outputs:
+      working_dir: ${{ steps.set_dir.outputs.dir }}
+    steps:
+      - name: Validate version format
+        run: |
+          if [[ ! "${{ inputs.version }}" =~ ^[0-9]+\.[0-9]+\.[0-9]+-rc\.[0-9]+$ ]]; then
+            echo "Error: Version must be in format X.Y.Z-rc.N (e.g., 1.2.0-rc.0)"
+            exit 1
+          fi
+
+      - name: Set working directory
+        id: set_dir
+        run: |
+          case "${{ inputs.package }}" in
+            schematic-js)
+              echo "dir=js" >> $GITHUB_OUTPUT
+              ;;
+            schematic-react)
+              echo "dir=react" >> $GITHUB_OUTPUT
+              ;;
+            schematic-components)
+              echo "dir=components" >> $GITHUB_OUTPUT
+              ;;
+            schematic-vue)
+              echo "dir=vue" >> $GITHUB_OUTPUT
+              ;;
+          esac
+
+  lint_test:
+    name: Lint and Test
+    needs: [validate]
+    uses: ./.github/workflows/reusable_lint_test.yml
+    with:
+      package: ${{ inputs.package }}
+      working_dir: ${{ needs.validate.outputs.working_dir }}
+
+  publish:
+    name: NPM Publish (RC)
+    runs-on: runs-on=${{ github.run_id }}/runner=4cpu-linux-x64/extras=s3-cache
+    permissions:
+      packages: write
+      contents: write
+    needs: [validate, lint_test]
+    steps:
+      - uses: runs-on/action@v2.0.3
+      - uses: actions/checkout@v6
+
+      - name: Set Node.js 24.x
+        uses: actions/setup-node@v6
+        with:
+          node-version: 24.x
+
+      - name: Yarn install
+        working-directory: ./${{ needs.validate.outputs.working_dir }}
+        run: yarn install
+
+      - name: Update package.json version
+        working-directory: ./${{ needs.validate.outputs.working_dir }}
+        run: |
+          npm version ${{ inputs.version }} --no-git-tag-version
+
+      - name: Build
+        working-directory: ./${{ needs.validate.outputs.working_dir }}
+        run: yarn build
+
+      - name: Set up .npmrc
+        run: touch .npmrc && echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+        working-directory: ./${{ needs.validate.outputs.working_dir }}
+
+      - name: Publish to NPM (RC tag)
+        if: ${{ !inputs.dry_run }}
+        working-directory: ./${{ needs.validate.outputs.working_dir }}
+        run: yarn publish --access public --tag rc
```

Comment:
Does this work with NPM? There are two possibles issues I have in mind:

1. I've found that NPM usually will throw an error if you try to reuse a tag (in this case, it looks like me like you'd be reusing a tag "rc")
2. When we tried to get release candidates working before, we couldn't figure out a way to make it so that release candidates are only installed with intent - i.e. if the most recent publish to NPM is a release candidate, and someone runs `yarn install mypackage` or `npm install mypackage`, they install the release candidate. Is there a way to prevent this?